vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -232,6 +232,7 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
     }
     HttpServerRequest request = context.request();
     if (!methods.isEmpty() && !methods.contains(request.method())) {
+      ((RoutingContextImplBase) context).setDefaultErrorCode(405);
```

Comment:
I'm wondering if the RFC says something about the order of status code returned.

Example (very marginal use case, but... who knows) : 

```java
router.get("/api/stuff").produces("application/json").handler(...);
```

```text
POST "/api/stuff"
  Accept: "text/html"
  Body: {}
```

Should it return 406 or 405 ? I didn't find anything in the RFC, just [That link re. Spring MVC](http://stackoverflow.com/questions/20288225/why-spring-gives-priority-to-405-http-error-over-406-http-error). What do you think Paulo ?